runtime/common/src/lib.rs

```diff
@@ -805,6 +805,34 @@ macro_rules! impl_config_traits {
                     ProxyType::Staking => matches!(c, RuntimeCall::ParachainStaking(..)),
                     #[cfg(not(feature = "parachain"))]
                     ProxyType::Staking => false,
+                    ProxyType::CreateMarket =>
+                        matches!(c,
+                            RuntimeCall::PredictionMarkets(zrml_prediction_markets::Call::create_market {..}) |
+                            RuntimeCall::PredictionMarkets(zrml_prediction_markets::Call::create_cpmm_market_and_deploy_assets {..})
+                        ),
+                    ProxyType::ReportOutcome =>
+                        matches!(c,
+                            RuntimeCall::PredictionMarkets(zrml_prediction_markets::Call::report {..})
+                        ),
+                    ProxyType::ProvideLiquidity =>
+                        matches!(c,
+                            RuntimeCall::Swaps(zrml_swaps::Call::pool_join {..}) |
+                            RuntimeCall::Swaps(zrml_swaps::Call::pool_join_with_exact_asset_amount {..}) |
+                            RuntimeCall::Swaps(zrml_swaps::Call::pool_join_with_exact_pool_amount {..}) |
+                            RuntimeCall::Swaps(zrml_swaps::Call::pool_exit {..}) |
+                            RuntimeCall::Swaps(zrml_swaps::Call::pool_exit_with_exact_asset_amount {..}) |
+                            RuntimeCall::Swaps(zrml_swaps::Call::pool_exit_with_exact_pool_amount {..})
+                        ),
+                    ProxyType::BuySellCompleteSets =>
+                        matches!(c,
+                            RuntimeCall::PredictionMarkets(zrml_prediction_markets::Call::buy_complete_set {..}) |
+                            RuntimeCall::PredictionMarkets(zrml_prediction_markets::Call::sell_complete_set {..})
+                        ),
```

Comment:
I think it would be useful to add `deploy_swap_pool_for_market`. Probably it makes sense to add a new proxy `DeployPool` for that.

Reply:
I think combine `DeployPool` and `Liquidity` into one, because they amount to the same thing. Also `deploy_swap_pool_and_additional_liquidity` is important. In fact, that's really the only call for deploying pools that we seem to use.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1038/commits/b80ae75c2da1bc7ce687cb5e2a5aa6d1aef2d37c